src/evolutionary_keras/optimizers.py

```diff
@@ -101,6 +102,17 @@ def get_shape(self):
         # TODO: eventually we should save here a reference to the layer and their
         # corresponding weights, since the nodes are the output of the layer
         # and the weights the corresponding to that layer
+
+        for layer in self.model.layers:
+            if layer.trainable_weights != []:
+                if isinstance(layer, Dense) != True:

```

Comment:
As we discussed earlier, a non-standard Keras can be used which will not be of type Dense and but which has the Dense structure.
I would either change the raising of the exception to a warning (so that the user the same msg as a warning saying "if this is wrong is problem") or add a `__init__` argument `skip_checks` and when explicitly set to True the checks are skipped.